Einige Bilder haben allen Kriterien beim download entsprochen, obwohl sie als "nicht verfügbar" gekennzeichnet waren. Diese Bilder werden nun gelöscht.

In [ ]:
import pandas as pd
import os

df = pd.read_csv("../all_csv_images/ocr_google_vision.csv")

df = df[df["ocr_text"] != "The image\nyou are\nrequesting does not exist\nor is no longer available.\nimgur.com"]


df_deleted = pd.read_csv("../all_csv_images/ocr_google_vision.csv")
df_deleted = df_deleted[df_deleted["ocr_text"] == "The image\nyou are\nrequesting does not exist\nor is no longer available.\nimgur.com"]

for index, row in df_deleted.iterrows():
    os.remove(row["relative_path"])


df.to_csv("../all_csv_images/filtered_file_ocr.csv",index=False)


Es gibt einige Memes die keinen Text enthalten. Jedoch wurden diese untersucht und es stellte sich heraus, dass der Reddit Titel des Postes als Text verwendet wurde. Aus diesem Grund wurden die leeren OCR-Texte mit dem Titel des Postes ersetzt.

In [3]:
empty_ocr_text_count = df['ocr_text'].isnull().sum()
print(f'Number of empty ocr_text: {empty_ocr_text_count}')

Number of empty ocr_text: 1086


In [ ]:
df = pd.read_csv('../all_csv_images/filtered_file_ocr.csv')

In [46]:
df.shape

(27293, 17)

In [47]:
df['ocr_text'].fillna(df['title'], inplace=True)

Einige Reddit Nutzer haben ihre Posts mehrfach gepostet. Dies ist ein Problem, da die Bilder dann mehrfach heruntergeladen werden. Aus diesem Grund werden alle Duplikate gelöscht.

In [48]:
duplicates = df[df.duplicated(['ocr_text'])]

In [49]:
duplicates.shape

(3918, 17)

In [50]:
df.drop_duplicates(subset=['ocr_text'],inplace=True)

In [51]:
df.shape

(23375, 17)

In [52]:
for index, row in duplicates.iterrows():
    os.remove(row["relative_path"])

In [53]:
df.to_csv("../all_csv_images/filtered_file_ocr_no_duplicates.csv",index=False)

Die verblienen Daten sind nun komplett gereinigt und können für die weitere Analyse verwendet werden.

Wir besitzen nun einen Datensatz mit 23375 Posts

In [54]:
df = pd.read_csv('../all_csv_images/filtered_file_ocr_no_duplicates.csv')
df.shape

(23375, 17)